Release 2.14.10 release notes (#11357)
PrefectHQ/prefect: RELEASE-NOTES.md

Creating app registration
Provisioning infrastructure... ━━━━━━━━╺━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  20% -:--:--App registration 'prefect-aci-push-pool-app' created successfully
Generating secret for app registration
Provisioning infrastructure... ━━━━━━━━━━━━━━━━╺━━━━━━━━━━━━━━━━━━━━━━━  40% 0:00:06Secret generated for app registration with client ID 'abc'
ACI credentials block 'aci-push-pool-credentials' created
Assigning Contributor role to service account...
Provisioning infrastructure... ━━━━━━━━━━━━━━━━━━━━━━━━╺━━━━━━━━━━━━━━━  60% 0:00:06Contributor role assigned to service principal with object ID 'xyz'
Creating Azure Container Instance
Provisioning infrastructure... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╺━━━━━━━  80% 0:00:04Container instance 'prefect-acipool-container' created successfully
Creating Azure Container Instance credentials block
Provisioning infrastructure... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Infrastructure successfully provisioned for 'my-work-pool' work pool!
Created work pool 'my-work-pool'!
```

> This marks a step forward in Prefect's Azure capabilities, offering users an efficient and streamlined process for leveraging Azure Container Instances in their workflow automation.

Reviewer: ```suggestion
This marks a step forward in Prefect's Azure capabilities, offering you an efficient and streamlined process for leveraging Azure Container Instances to execute their workflows.
```

│         - Create Azure Container Instance 'aci-push-pool-container' in resource group prefect-aci-push-pool-rg │
│                                                                                                                │
│     Updates in Prefect workspace                                                                               │
│                                                                                                                │
│         - Create Azure Container Instance credentials block aci-push-pool-credentials                          │
│                                                                                                                │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
Proceed with infrastructure provisioning? [y/n]: y
...
```

This PR bolsters support for efficient work pool management across diverse cloud environments, delivering a tool for seamless infrastructure setup.

See the following pull request for implementation details:
- https://github.com/PrefectHQ/prefect/pull/11341

### Enhancements

> - Added a `suspend_flow_run` method to suspend a flow run. — https://github.com/PrefectHQ/prefect/pull/11291

Reviewer 2: ```suggestion
- Added a `suspend_flow_run` method to suspend a flow run — https://github.com/PrefectHQ/prefect/pull/11291
```

Author: Some of these comments were dropped on my latest push, but I followed your suggestions 👍🏼

- Limited the displayed work pool types when `--provision-infra` is used to only show supported work pool types - https://github.com/PrefectHQ/prefect/pull/11350
- Added the ability to publish `Infrastructure` blocks as work pools — https://github.com/PrefectHQ/prefect/pull/11180
- Added the ability to publish `Process` blocks as work pools — https://github.com/PrefectHQ/prefect/pull/11346
- Added a Prefect Cloud event stream subscriber — https://github.com/PrefectHQ/prefect/pull/11332

> - Added the data model and methods for `FlowRunInput` — https://github.com/PrefectHQ/prefect/pull/11342

Reviewer 2: "enable storage of key/value information associated with a flow run"

- Added the data model and methods for `FlowRunInput` — https://github.com/PrefectHQ/prefect/pull/11342
- Made `FlowRunInput.flow_run_id` a foreign key to `flow_run.id` — https://github.com/PrefectHQ/prefect/pull/11352

### Fixes
- Fixed the `read_logs` return type to be `List[Log]` — https://github.com/PrefectHQ/prefect/pull/11303

> - Fixed an issue where submitting a task and then pausing the flow run without using `reschedule=True` would cause the flow run to become stuck in the `Paused` state — https://github.com/PrefectHQ/prefect/pull/11284

Reviewer 2: I think this is too long

### Documentation
- Combined troubleshooting pages — https://github.com/PrefectHQ/prefect/pull/11288
- Added Google Cloud Run V2 option to Serverless guide — https://github.com/PrefectHQ/prefect/pull/11304
- Added `suspend_flow_run` to flows documentation — https://github.com/PrefectHQ/prefect/pull/11300
- Added `work queues` tag to work pools concept page — https://github.com/PrefectHQ/prefect/pull/11320
- Added missing Python SDK CLI items to the docs — https://github.com/PrefectHQ/prefect/pull/11289
- Clarified SCIM + service accounts handling — https://github.com/PrefectHQ/prefect/pull/11343

> - Updated the work pool concept document — https://github.com/PrefectHQ/prefect/pull/11331

Reviewer 2: These should be in present tense

Author: These have been present-tense-ifiied
